virtio-devices/src/block.rs

```diff
@@ -389,12 +389,7 @@ impl BlockEpollHandler {
 
                 (VIRTIO_BLK_S_OK as u8, result as u32)
             } else {
-                error!(
-                    "Request failed: {:x?} {:?}",
-                    request,
-                    io::Error::from_raw_os_error(-result)
-                );
-                return Err(Error::AsyncRequestFailure);
+                (VIRTIO_BLK_S_IOERR as u8, result as u32)
```

Comment:
I don't think this is correct - you are casting the `i32` of negative value to `u32`.

Also, let's keep the error message here so that we have something from our log when such errors happen